master/static/srv/entrypoint.sh

```diff
@@ -7,6 +7,16 @@ WORKING_DIR="/run/determined/workdir"
 STARTUP_HOOK="startup-hook.sh"
 export PATH="/run/determined/pythonuserbase/bin:$PATH"
 
+# If HOME is not explicitly set for a container, libcontainer (Docker) will
+# try to guess it by reading /etc/password directly, which will not work with
+# our linss_determined extension (or any user-defined extension in a container).
+# The default is "/", but HOME must be a writable location for distributed
+# training, so we try to query the user system for a valid HOME, or default to
+# the working directory otherwise.
+if [ "$HOME" = "/" ] ; then
+    export HOME="$(getent passwd "$(whoami)" | cut -d: -f6)" || export HOME="$WORKING_DIR"
+fi
```

Comment:
A simpler alternative to this would be to set the HOME environment variable to /run/determined/workdir in trial.go when we launch the container, but I figured this would play nicer with images that have the user baked into the image.

I'm quite open to changing this.